03_launch_mgmt_cluster.sh

```diff
@@ -252,7 +252,7 @@ function make_bm_hosts() {
 function apply_bm_hosts() {
   pushd "${BMOPATH}"
   list_nodes | make_bm_hosts > "${WORKING_DIR}/bmhosts_crs.yaml"
-  if [ "${EPHEMERAL_CLUSTER}" != "tilt" ]; then
+  if [[ -n "$(list_nodes)" ]]; then
```

Comment:
What if we `apply_bm_hosts` is called from some other place , the condition should not be changed IMO? Whats the motivation of changing this?

Reply:
The condition `${EPHEMERAL_CLUSTER}" != "tilt"` is already checked at the only site where `apply_bm_hosts` is called in the codebase (https://github.com/metal3-io/metal3-dev-env/blob/f089833114e5a722f6ed69e42eea660353168f4b/03_launch_mgmt_cluster.sh#L471-L474), so this function will never be called for a tilt cluster. Removing the check slightly improves the intent of the function. When I looked at it, I wondered why bmhosts_crs.yaml would be generated but not applied for a tilt cluster. But it turns out in reality, it is neither generated nor applied for a tilt cluster.